### PR TITLE
optimize: repeat pk in after image sql for INSERT_ON_DUPLICATE_UPDATE

### LINKED ISSUE #4955: create afterimages sql repeat pk
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
构建后镜像sql生成没有进行去重等处理,在前镜像生成sql后,导致后镜像复用前镜像的sql,并往里面放主键的时候造成重复
image

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- Seata version:  
- OS :
- Others:

### PR DESCRIPTION
**Pull request by renliangyu857**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅱ. Does this pull request fix one issue?

fixes #4955

### CODE REVIEW

**Review by a364176773 — COMMENTED** [review]
Why don't you record against unique indexes?
为什么不对唯一索引进行记录？

**Comment by renliangyu857** [comment]
后置镜像是否可以只通过主键查询，无需带上唯一健？待验证

**Review by a364176773 — APPROVED** [review]
LGTM

**Comment by a364176773** [comment]
@slievrly @jsbxyyx @caohdgege @l81893521 PTAL

**Review by jsbxyyx — APPROVED** [review]
LGTM

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/mysql/MySQLInsertOnDuplicateUpdateExecutor.java [hunk 1]
```diff
@@ -317,6 +329,9 @@ public String buildImageSQL(TableMeta tableMeta) {
                             }
                             break;
                         }
+                        if ("PRIMARY".equalsIgnoreCase(k)) {
+                            primaryKeysInBeforeImageSql.add(columnName);
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/mysql/MySQLInsertOnDuplicateUpdateExecutor.java:333]: there may be an escape character issue here. Keywords require escape characters, such as' order '.

**Pull request merged**